invenio_rdm_records/mappings/v6/records/record-v1.0.0.json

```diff
@@ -273,11 +270,8 @@
                 "name": {
                   "type": "text"
                 },
-                "identifier": {
-                  "type": "keyword"
-                },
-                "scheme": {
-                  "type": "keyword"
+                "identifiers": {
+                  "type": "object"
```

Comment:
minor: how many key-value pairs do we expect. Since keys will be dynamically created we could have a case of mapping explosion.

Reply:
Very few. In fact right now the marshmallow schema only accepts ROR identfifiers for affiliations so it is 0 or 1.

Is it more problematic that a large number of identifiers are in object mapping rather than array mapping?

Reply:
It depends on the amount of keys... If you have for example:

```
{
 id_type_one: id_value,
 id_type_two: id_value
}
```

Means that **Every** document will have said key, it might be that the value is nulled, but the key will be there. If there are two or few keys is not a problem. However, if it was the case that there were 1000 possible keys every single document would have:

```
{
 id_one: value,
 ...
 id_one_thousand: value
}
```

However, with array the amount of keys would always be two:
```
[ { id: id_one, value: value}, {id: id_one_thousand, value:value} ]
```

I understand from what you said it is not our case. However, if this was to grow a lot in keys it can have impact on search performance and even produce run time errors when indexing since the amount of keys is limited at mapping definition level. More info [here](https://www.elastic.co/guide/en/elasticsearch/reference/master/mapping.html#mapping-limit-settings)

Reply:
> Means that Every document will have said key, it might be that the value is nulled, but the key will be there.

Oh interesting. So if:

1- index document A with {identifiers: {}}
2- index document B with {identifiers: {ror: blabalba}}

ES reindexes document A with {identifiers: {ror: null}} and so on for each other key in identifiers?

Reply:
Yes, not fully sure if the default value is "", null or what exactly. Since ES is Java I would say sort of `null`. But yes...